include/revng/Support/IRHelpers.h

```diff
@@ -1363,6 +1363,14 @@ inline unsigned getMemoryAccessSize(llvm::Instruction *I) {
   return getPointeeSize(getPointer(I));
 }
 
+inline llvm::Value *hasDynamicSymbol(llvm::BasicBlock *BB) {
```

Comment:
* Rename to `getDynamicSymbol`.
* Can we return directly the `StringRef` to the symbol name? Seems like the right thing to do.

One day, we'll need to split off `IRHelpers.h`.

Reply:
1) OK.
2) I think it makes more sense to return a `Value*`, easier to operate with, and there is already a dedicated method `extractFromConstantStringPtr` in `IRHelpers.h` responsible for returning the symbol name, if `Value*` exists.

Reply:
There's only one usage of this function and the first thing you do is extracting the string. How is returning a `Value *` easier to operate?
And I think this is basically what every user of this function will want: get the name so they can go to the model and extract information about the symbol.

Reply:
OK, so you're saying to make `getDynamicSymbol` directly call `extractFromConstantStringPtr`, right? Should `extractFromConstantStringPtr` still exist at this point or should we drop it?

Reply:
I think that's either used or can be used in other situations where we have constant strings and we want to get to the data easily (e.g., assembly). Let's keep it.

Reply:
In `include/revng/EarlyFunctionAnalysis/CallEdge.h`, now there's `inline model::TypePath getPrototype(const model::Binary &Binary, MetaAddress CallerFunctionAddress, MetaAddress CallerBlockAddress, const efa::CallEdge &Edge);` (was previously [this one](https://github.com/revng/revng/blob/develop/include/revng/Model/Binary.h#L131), now moved in `efa` namespace). Are we sure we still want to do another specification of this function?

Reply:
Please don't mark as resolved discussions where you're asking a question.

If someones uses `hasDynamicSymbol`, it needs to have the characteristics we discussed... if no one uses it, I guess it can be dropped.

Reply:
Sorry, there must have been a mistake, this issue has been fixed. The question I asked is for the review below, not for this one. I copy and paste my comment below.

Reply:
ACK